src/map/skill.cpp

```diff
@@ -15812,15 +15831,32 @@ static int32 skill_dance_overlap_sub(struct block_list* bl, va_list ap)
 	return 1;
 }
 
-//Does the song/dance overlapping -> dissonance check. [Skotlex]
-//When flag is 0, this unit is about to be removed, cancel the dissonance effect
-//When 1, this unit has been positioned, so start the cancel effect.
-int32 skill_dance_overlap(struct skill_unit* unit, int32 flag)
+/**
+ * Does the song / dance overlapping->dissonance check.
+ * @param unit The unit that is about to be removed or was placed
+ * @param flag Whether unit is about to be removed or was placed
+ *   When flag is OVERLAP_REMOVE, this unit is about to be removed, remove the dissonance effect from applicable units on the cell
+ *   When flag is OVERLAP_SET, this unit has been positioned, start the dissonance effect for applicable units on the cell
+ * @return Number of skill units that overlap with current skill unit and meet the overlap criteria (excluding current skill unit)
```

Comment:
Number is never really used on any call.
Consider making the function void instead.

Reply:
I was thinking it might be a useful feature for custom / future implementations. Also otherwise one of the possible flag values would do nothing.